source/lac/trilinos_sparsity_pattern.cc

```diff
@@ -793,17 +793,26 @@ namespace TrilinosWrappers
         if (nonlocal_graph->IndicesAreGlobal() == false &&
             nonlocal_graph->RowMap().NumMyElements() > 0)
           {
-            // insert dummy element
+            // Insert dummy element at (row, row) that corresponds to row 0 in
+            // local index counting.
+            //
+            // Note: This call to InsertGlobalIndices can fail with error code
+            // -2 if the row is not available locally. This is okay though as
+            // it will exist for the owner of that row already. Do not check
+            // the return value here.
             TrilinosWrappers::types::int_type row =
-              nonlocal_graph->RowMap().MyGID(0);
+              nonlocal_graph->RowMap().GID64(0);
```

Comment:
So does the function return `-2` then and puts that into `row`? How is using that value in the next line going to work then?

Reply:
Oh, forget it. It's the *following* line that errors out. 

A better approach would check the error code and use `ierr==0 || ierr==-2`.

Reply:
But we should not have any return value `-2` neither here nor in the next line, because the whole block is only entered in case there is at least one local line present, and we pick the zeroth such row. So in any case, we do a valid operation here. Or do I misunderstand something here?

Reply:
Wait, I think we should put `TrilinosWrappers::global_index(nonlocal_graph->RowMap(), 0)` here, with the `global_index` function defined in `trilinos_index_access.h`, to make this work for both 32 and 64 bit integers.

Reply:
> A better approach would check the error code and use `ierr==0 || ierr==-2`

No. I have observed return values of -2,0,1,2 so far. >= apparently is success (but denotes that the entries already exist or something). I don't think it makes sense to check things this way.

Reply:
> Or do I misunderstand something here?

I don't know, to be honest.

> Wait, I think we should put TrilinosWrappers::global_index(nonlocal_graph->RowMap(), 0) here
> to make this work for both 32 and 64 bit integers.

To me it looks like GID64 is defined in 32 and 64 bit mode, at least in my version of trilinos. I will test that.